python/ray/includes/libcoreworker.pxi

```diff
@@ -46,3 +46,48 @@ cdef class ProfileEvent:
         # Deleting the CProfileEvent will add it to a queue to be pushed to
         # the driver.
         self.inner.reset()
+
+
+cdef class ObjectLocation:
+    """Cython wrapper class of C++ `ray::ObjectLocation`."""
+    cdef:
+        shared_ptr[CObjectLocation] inner_
+
+    @staticmethod
+    cdef make(const shared_ptr[CObjectLocation]& object_location):
+        cdef ObjectLocation self = ObjectLocation.__new__(ObjectLocation)
+        self.inner_ = object_location
+        return self
+
+    def primary_node_id(self):
+        """Return the ID of the node has the primary copy of the object.
+           Return None if the object is pending, inlined or evicted."""
+        id = NodeID(self.inner_.get().GetPrimaryNodeID().Binary())
+        return None if id.is_nil() else id
+
+    def object_size(self):
+        """Return the size of the object"""
+        return self.inner_.get().GetObjectSize()
+
+    def node_ids(self):
+        """Return the IDs of the nodes that this object
+           appeared on or was evicted by"""
+        cdef c_vector[CNodeID] c_node_ids = self.inner_.get().GetNodeIDs()
+        result = []
+        for c_node_id in c_node_ids:
+            result.append(NodeID(c_node_id.Binary()))
+        return result
+
+    def is_spilled(self):
+        """Return wether this object has been spilled"""
+        return self.inner_.get().IsSpilled()
+
+    def spilled_url(self):
+        """Return the spilled location, None if not spilled"""
+        return self.inner_.get().GetSpilledURL() if self.is_spilled() else None
+
+    def spilled_node_id(self):
+        """Return the node ID which spilled the object. None if the object
+           is not spilled or was spilled to distributed external storage"""
+        id = NodeID(self.inner_.get().GetSpilledNodeID().Binary())
```

Comment:
We can return the Hex representation as a string instead of the Cython NodeID. Btw, we should probably do the same for runtime_env (avoid exposing Cython types).